num_connections: 4
connector_num: [2 3 4 5];
